src/Fmt_ast.ml

```diff
@@ -2968,19 +2980,21 @@ and fmt_module_type c ({ast= mty} as xmty) =
             $ fmt "@;<1 2>-> " $ Option.call ~f:blk.pro )
       ; epi= Some (Option.call ~f:blk.epi $ Cmts.fmt_after c.cmts pmty_loc)
       }
-  | Pmty_with (mt, wcs) ->
-      let {opn; pro; psp; bdy; cls; esp; epi} =
-        fmt_module_type c (sub_mty ~ctx mt)
-      in
+  | Pmty_with _ ->
+      let wcs, mt = sugar_mod_with c (sub_mty ~ctx mty) in
+      let {opn; pro; psp; bdy; cls; esp; epi} = fmt_module_type c mt in
       { empty with
         bdy=
           hvbox 0
             (wrap_if parens "(" ")"
                ( opn $ Option.call ~f:pro $ psp $ bdy $ cls $ esp
                $ Option.call ~f:epi
-               $ list_fl wcs (fun ~first ~last:_ wc ->
-                     fmt_or first "@ with" "@;<1 1>and"
-                     $ fmt_with_constraint c ctx wc ) ))
+               $ list_fl (List.rev wcs)
```

Comment:
The other `sugar_` functions don't return reversed results, so it might be clearer to move this `List.rev` into `sugar_mod_with`